B2C Implementation Guide (#7310)
auth0/docs: articles/architecture-scenarios/b2c/planning.md

---
title: Planning Your Consumer Identity and Access Management Integration
description: Why it's important to plan your Consumer Identity and Access Management Implementation prior to beginning the integration
toc: false
topics:
    - b2c
    - ciam
contentType: concept
useCase:
  - implementation
---
# Planning Your Consumer Identity and Access Management Integration

In our experience, customers who establish clear objectives and align those objectives to requirements early on benefit the most from their integration with Auth0. Regardless of the project type, we recommend that you plan your project using a phased approach across multiple workstreams. In most cases, we see the most success with those adopting a three-phase project structure.

::: note

> Identifying your primary objectives early on during the project will help you focus on the specifics important to building out your solution. For instance, if your primary objective is to avoid disruption to end users, then you should not adopt a "big bang" approach when integrating with Auth0.

Reviewer: ```suggestion
Identifying your primary objectives early on during the project will also help you focus on the specifics important to building out your solution. For instance, if your primary objective is to avoid disruption to end users, then you should not adopt a "big bang" approach when integrating with Auth0.
```

:::

> We also think that customers who adopt an iterative release process make better progress. For example, let's say that you have 3-4 applications you want to integrate with Auth0. Rather than integrating all at once and shipping only when that's done, you might consider completing the integration for one application before moving on to the next. This way, your teams can benefit from the experience of working with Auth0 and leverage this experience to help increase the velocity of each iteration.

Reviewer: When putting the original B2C Welcome together, the consensus was that the language used should be as prescriptive and definitive as possible. So perhaps something like this would work better:
```suggestion
The evidence we've collected suggests that customers who also adopt an iterative release process make better progress. For example, let's say that you have 3-4 applications you want to integrate with Auth0. Rather than integrating all at once and shipping only when that's done, you might consider completing the integration for one application before moving on to the next. This way, your teams can benefit from the experience of working with Auth0 and leverage this experience to help increase the velocity of each iteration.
```

::: note-warning

> While adopting an iterative release process will improve your time to market when integrating multiple applications, there are additional considerations for you to review if you want to support Single Sign-On (SSO).

Reviewer: ```suggestion
While adopting an iterative release process will improve your time to market when integrating multiple applications, there are additional considerations for you to review if you want to support [Single Sign-On (SSO)](#).
```

:::

> There may be other groups within your organization who have worked with Auth0 and from whom you can go to for first-hand experience. It's not uncommon for us to work with different parts of the same company at different points of time. We may be able to help you identify these groups prior to the beginning of your project, which might help you as you structure your initial project plans.

Reviewer: ```suggestion
In addition, there may be other groups within your organization who have worked with Auth0 and from whom you can go to for first-hand experience - it's not uncommon for us to work with different parts of the same company at different points in time. We may be able to help you identify these groups prior to the beginning of your project, which might help you as you structure your initial project plans.
```
